### PR TITLE
Basic main menu

### LINKED ISSUE #5: Create a basic main menu
Probably a temporary main menu, with no art/graphic design, just some buttons that lead to placeholder pages or are disabled?

- Play
- Achievements
- Settings
- About/credits?

Probably done by putting a bunch of https://api.flutter.dev/flutter/material/RaisedButton-class.html on main page, using https://api.flutter.dev/flutter/widgets/ListView-class.html for layout?

The old home screen should be where the new play button leads to.

### PR DESCRIPTION
**Pull request by JonMasson**
- Split up main.dart into multiple files (following [this ](https://medium.com/hackernoon/scalable-app-structure-in-flutter-dad61a4bc389)folder structure)
- New home screen with 3 buttons
- Implemented 'Play' button, the 'Achievements' and 'Settings' buttons are disabled
- Old home screen is now the play screen

Main menu

Play page

Closes #5

### CODE REVIEW

File: lib/screens/main_menu.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+import 'package:treasure_of_the_high_seas/screens/play.dart';
+
+class MainMenuPage extends StatelessWidget {
+  MainMenuPage({Key key}) : super(key: key);
```
Comment (JonMasson) [inline lib/screens/main_menu.dart:5]: Not sure what this does or if it's still necessary

File: lib/screens/main_menu.dart [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Treasure of the High Seas'),
+      ),
+      body: Center(
+        child: ButtonTheme(
+          height: 100,
```
Comment (JonMasson) [inline lib/screens/main_menu.dart:15]: Not sure about putting in a fixed height, not very responsive on different sized screens

File: lib/screens/main_menu.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          minWidth: (MediaQuery.of(context).size.width) * 0.8,
```
Comment (JonMasson) [inline lib/screens/main_menu.dart:16]: Couldn't find a clean way to get 80% width on buttons...

File: lib/screens/main_menu.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      body: Center(
+        child: ButtonTheme(
+          height: 100,
+          minWidth: (MediaQuery.of(context).size.width) * 0.8,       
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            mainAxisSize: MainAxisSize.min,
+            children: <Widget>[
+              Padding(
+                padding: const EdgeInsets.all(10.0),
```
Comment (JonMasson) [inline lib/screens/main_menu.dart:22]: Wasn't sure how to space out the buttons properly, also this padding is duplicated for all 3 buttons

File: lib/screens/main_menu.dart [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                child: RaisedButton(
+                  onPressed: () {
+                    Navigator.push(
+                      context,
+                      MaterialPageRoute(builder: (context) => PlayPage(title: 'Play Page')),
+                    );
+                  },
+                  color: Colors.green,
```
Comment (JonMasson) [inline lib/screens/main_menu.dart:30]: Colour should come from App theme somehow?

File: lib/screens/main_menu.dart [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                  child: const Text(
+                    'Play',
+                    style: TextStyle(fontSize: 30)
```
Comment (JonMasson) [inline lib/screens/main_menu.dart:33]: Feels wrong to have text size set like this, and duplicated on each button, not sure how else to do it

**Review by kilikfan — APPROVED** [review]
Looks good to me.

File: lib/screens/main_menu.dart (hunk 1 above)
Comment (ghost) [inline lib/screens/main_menu.dart:5]: Do we need this constructor? Seems odd to have the optional param when we're not using it?

File: lib/screens/main_menu.dart [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+              Padding(
+                padding: const EdgeInsets.all(10.0),
+                child: RaisedButton(
+                  onPressed: null,
+                  child: const Text(
+                    'Settings',
+                    style: TextStyle(fontSize: 30)
+                  ),
+                ),
+              )
```
Comment (ghost) [inline lib/screens/main_menu.dart:56]: Could pull this out as a `menu_button.dart` maybe? Which could take the button text and `onPressed` as arrrrgs

File: lib/screens/main_menu.dart (hunk 1 above)
Comment (ghost) [reply lib/screens/main_menu.dart:5]: We can definitely get rid of it :+1:

File: lib/screens/main_menu.dart (hunk 3 above)
Comment (ghost) [reply lib/screens/main_menu.dart:16]: I think using `MediaQuery` is a pretty standard pattern for doing stuff like this - seems fine to me

File: lib/screens/main_menu.dart (hunk 4 above)
Comment (ghost) [reply lib/screens/main_menu.dart:22]: RE: duplication, yeah - we could pull out everything wrapped by the `Padding` into a reusable `menu_button` component

**Pull request merged**